lib/main_common.pm

```diff
@@ -456,6 +649,533 @@ sub boot_hdd_image {
+        loadtest "installation/dud_addon";
+    }
+    if (is_sle && sle_version_at_least('15')) {
+        loadtest "installation/accept_license" if get_var('HASLICENSE');
+    }
+    if (get_var('IBFT')) {
+        loadtest "installation/iscsi_configuration";
+    }
+    if (check_var('ARCH', 's390x')) {
+        if (check_var('BACKEND', 's390x')) {
+            loadtest "installation/disk_activation";
+        }
+        elsif (!sle_version_at_least('12-SP2')) {
+            loadtest "installation/skip_disk_activation";
+        }
+    }
+    if (get_var('ENCRYPT_CANCEL_EXISTING') || get_var('ENCRYPT_ACTIVATE_EXISTING')) {
+        loadtest "installation/encrypted_volume_activation";
+    }
+    if (get_var('MULTIPATH')) {
+        loadtest "installation/multipath";
+    }
+    if (is_opensuse && noupdatestep_is_applicable() && !get_var("LIVECD")) {
+        loadtest "installation/installation_mode";
+    }
+    if (!get_var("LIVECD") && get_var("UPGRADE")) {
+        loadtest "installation/upgrade_select";
+        if (check_var("UPGRADE", "LOW_SPACE")) {
+            loadtest "installation/disk_space_fill";
+        }
+        loadtest "installation/upgrade_select_opensuse" if is_opensuse;
+    }
+    if (is_sle) {
+        # SCC registration is not required in media based upgrade since SLE15
+        unless (sle_version_at_least('15') && get_var('MEDIA_UPGRADE')) {
+            if (check_var('SCC_REGISTER', 'installation')) {
+                loadtest "installation/scc_registration";
+            }
+            else {
+                loadtest "installation/skip_registration" unless check_var('SLE_PRODUCT', 'leanos');
+            }
+        }
+        if (is_sles4sap and !sle_version_at_least('15')) {
+            loadtest "installation/sles4sap_product_installation_mode";
+        }
+        if (get_var('MAINT_TEST_REPO')) {
+            loadtest 'installation/add_update_test_repo';
+        }
+        loadtest "installation/addon_products_sle";
+    }
+    if (noupdatestep_is_applicable()) {
+        # Krypton/Argon disable the network configuration stage
+        if (get_var("LIVECD") && !is_krypton_argon) {
+            loadtest "installation/livecd_network_settings";
+        }
+        #system_role selection during installation was added as a new feature since sles12sp2
+        #so system_role.pm should be loaded for all tests that actually install to versions over sles12sp2
+        #no matter with or without INSTALL_TO_OTHERS tag
+        if (
+            is_sle
+            && (   check_var('ARCH', 'x86_64')
+                && sle_version_at_least('12-SP2')
+                && is_server()
+                && (!is_sles4sap() || is_sles4sap_standard())
+                && (install_this_version() || install_to_other_at_least('12-SP2'))
+                || sle_version_at_least('15')))
+        {
+            loadtest "installation/system_role";
+        }
+        if (is_sles4sap() and sle_version_at_least('15')) {
```

Comment:
Why was `and check_var('SYSTEM_ROLE', 'default')` dropped from the condition? In SLES4SAP, when any of the SLES System Roles are selected, the installer goes directly to partitioning instead of the SLES4SAP Product Installation Mode screen.

Reply:
Faling test: https://openqa.suse.de/tests/1492176#step/sles4sap_product_installation_mode/2